File: ios/Sources/EdgeFrameworks/Core/EdgeKnowledgeManagementModels.swift
public struct EdgeKnowledgeCollectionSummary:
    Equatable,
    Sendable
{
    public let collection: EdgeKnowledgeCollection
    public let sourceCount: Int
    public let documentCount: Int
    public let lastIndexedAtMilliseconds: Int64?

    public init(
        collection: EdgeKnowledgeCollection,
        sourceCount: Int,
        documentCount: Int,
        lastIndexedAtMilliseconds: Int64?
    ) {
        self.collection = collection
        self.sourceCount = sourceCount
        self.documentCount = documentCount
        self.lastIndexedAtMilliseconds =
            lastIndexedAtMilliseconds
    }
}

public enum EdgeKnowledgeManagerError:
    Error,
    Equatable,
    Sendable
{
    case collectionNotFound(String)
}

File: android/edge-frameworks-core/src/main/kotlin/io/github/rajibsarwar/edgeframeworks/EdgeKnowledgeManagementModels.kt
package io.github.rajibsarwar.edgeframeworks

data class EdgeKnowledgeCollectionSummary(
    val collection: EdgeKnowledgeCollection,
    val sourceCount: Int,
    val documentCount: Int,
    val lastIndexedAtMilliseconds: Long?
)

sealed class EdgeKnowledgeManagerException(
    message: String
) : Exception(message) {
    data class CollectionNotFound(
        val collectionId: String
    ) : EdgeKnowledgeManagerException(
        "Knowledge collection not found: $collectionId"
    )
}
